src/instructlab/model/chat.py

```diff
@@ -789,6 +536,157 @@ def start_prompt(
         self._update_conversation(response_content.plain, "assistant")
 
 
+def chat_model(
+    ctx,
+    question,
+    model,
+    context,
+    session,
+    quick_question,
+    max_tokens,
+    endpoint_url,
+    api_key,
+    tls_insecure,
+    tls_client_cert,
+    tls_client_key,
+    tls_client_passwd,
+    model_family,
+    serving_log_file,
+    temperature,
+):
+    """Runs a chat using the modified model"""
+    # pylint: disable=import-outside-toplevel
+    # First Party
+    from instructlab.model.backends.common import is_temp_server_running
+
+    max_ctx_size = None
+    backend_type = ctx.obj.config.serve.server.backend_type
+    serve_server = ctx.obj.config.serve.server
+    users_endpoint_url = cfg.get_api_base(serve_server.host, serve_server.port)
+
+    # we prefer the given endpoint when one is provided, else we check if the user
+    # is actively serving something before falling back to serving our own model
+    backend_instance = None
+    if endpoint_url:
+        api_base = endpoint_url
+    elif is_openai_server_and_serving_model(
+        users_endpoint_url,
+        api_key,
+        http_params={
+            "tls_client_cert": tls_client_cert,
+            "tls_client_key": tls_client_key,
+            "tls_client_passwd": tls_client_passwd,
+            "tls_insecure": tls_insecure,
+        },
+    ):
+        api_base = users_endpoint_url
+        if serving_log_file:
+            logger.warning(
+                "Setting serving log file (--serving-log-file) is not supported when the server is already running"
+            )
+        max_ctx_size = ctx.obj.config.serve.server.current_max_ctx_size
+    else:
+        # If a log file is specified, write logs to the file
+        root_logger = logging.getLogger()
+        if serving_log_file:
+            log.add_file_handler_to_logger(root_logger, serving_log_file)
+
+        ctx.obj.config.serve.llama_cpp.llm_family = model_family
+        backend_instance = backends.select_backend(
+            ctx.obj.config.serve,
```

Comment:
same here, this might be kind of tricky. We might want to scope future work to remove the dependency on the entire cfg to select a backend